src/lib.rs

```diff
@@ -323,14 +323,27 @@ impl ClientHandle {
                     h = Some(inner);
                     info = Some(server_info);
                 }
-                Ok(Packet::Exception(e)) => return Err(Error::Server(e)),
-                Err(e) => return Err(Error::Io(e)),
-                _ => return Err(Error::Driver(DriverError::UnexpectedPacket)),
+                Ok(Packet::Exception(e)) => {
+                    log::error!("[hello] <- Server error: {:?}", e);
+                    return Err(Error::Server(e));
+                }
+                Err(e) => {
+                    log::error!("[hello] <- IO error: {:?}", e);
+                    return Err(Error::Io(e));
+                }
+                p => {
+                    log::error!("[hello] <- Unexpected packet received: {:?}", p);
+                    return Err(Error::Driver(DriverError::UnexpectedPacket));
+                }
             }
         }
 
         self.inner = h;
-        self.context.server_info = info.unwrap();
+        if let Some(server_info) = info {
+            self.context.server_info = server_info;
+        } else {
+            warn!("Server info is not received");
```

Comment:
Should this be an error? This should not happen under normal condition, right?

Reply:
Changed it to return an error. This happens because the server connection times out.

Reply:
@ncblakely Ready for review again :)